libs/eavmlib/src/uart.erl

```diff
@@ -88,17 +88,15 @@ warn_deprecated(OldKey, NewKey) ->
 validate_peripheral(I) when is_integer(I) ->
     io:format("UART: deprecated integer peripheral is used.~n"),
     I;
-validate_peripheral([$u, $a, $r, $t | N] = Value) ->
```

Comment:
I don't think we need to change to upper case to fix the problem here, or am I missing something?

Reply:
Oh really? On the Nif's side, the value of `peripheral` option is compared to uppercased `"UART0"`, `"UART1"` and `"UART2"` using `strcmp`. So I thought we have to compare it with uppercased string on the Erlang side too.
https://github.com/atomvm/AtomVM/blob/9be487f156a712865246b30036b8fc58907dd61a/src/platforms/esp32/components/avm_builtins/uart_driver.c#L203-L216

Reply:
My apologies, this does indeed look correct. Not sure how long this has gone unnoticed, but good catch!